server/services/userService.ts

```diff
@@ -27,6 +52,15 @@ export default class UserService {
     return { ...user, caseloads, displayName: StringUtils.convertToTitleCase(user.name) }
   }
 
+  /*
+   * A helper which presents a consistent language string across controllers
+   * And which factors in the user may no longer be an entity
+   */
+  async getFullNameFromUsername(userToken: Express.User['token'], username: User['username']): Promise<string> {
+    const user = await this.checkUserExistsFromUsername(userToken, username)
```

Comment:
I don't think there's a particular rush to get this in - I agree that it would be good to DRY this up to reduce the chance of people getting confused over which method to use.

Reply:
> So I think we could probably DRY this up by adding a boolean argument to `getUserFromUsername` which determines whether the method errors on 404
> 
> ```ts
>   async getUserFromUsername(userToken: Express.User['token'], username: User['username'], errorOn404?: boolean = false): Promise<User> {
>       [...]
> 
>       if (knownError.status === 404) {
>         if (errorOn404) {
>           return null
>         } 
> 
>         throw createError(knownError.status, `User with username ${username} not found.`)
>       }
> 
>       [...]
>   }
> ```
> 
> This would mean we wouldn't need the `checkUserExistsFromUsername` method, and we can still use `getUserFromUsername` as before, but with the boolean set to `true` in `getFullNameFromUsername`. Then we can hopefully just add a condition to the `getUserFromUsername` tests to cover this
> 
> What do you think @andrewcharnley?

I consider it to be an anti-pattern, in FP approach one function does one thing well.

I was 50/50 on whether to rewrite the original function to change the behaviour but decided against it for the reason that the original decision on propagating the error and not going any further had a reasonable chance at being a business led one. If not, it was T.D, and a second quick task can now be to kill it. I wouldn't say there's a right or wrong here as both approaches will end up in the same place.

Small comments like those above the functions are to differentiate the distinction, they are the 'hints and tips' which make developers lives easier.

Reply:
The original function is indeed still used and is a good example where behaviour may differentiate.

https://github.com/ministryofjustice/hmpps-accredited-programmes-ui/blob/302102f8b20370f36a7490fa2fbe008b683a7020/server/services/userService.ts#L48